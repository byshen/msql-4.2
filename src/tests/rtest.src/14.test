#
# Test the use of resorved words as idents
#

create table table(int int, char char(5), index uint)\p\g
create table unique(key int, date date)\p\g
create table updates(update text(10), update_time time)\p\g

insert into unique values (1, '1-Jan-1001')\p\g
insert into unique values (2, '2-Feb-2002')\p\g
select * from unique where date > '1-Jan-999' \p\g

#
# Test the new offset clause

select first_name, last_name from staff order by last_name \p\g
select first_name, last_name from staff order by last_name offset 2\p\g
select first_name, last_name from staff order by last_name limit 1 offset 3 \p\g
