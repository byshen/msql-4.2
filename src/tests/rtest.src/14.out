
Welcome to the miniSQL monitor.  Type \h for help.


mSQL >     -> 
Query buffer
------------

create table table(int int, char char(5), index uint)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
create table unique(key int, date date)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
create table updates(update text(10), update_time time)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into unique values (1, '1-Jan-1001')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into unique values (2, '2-Feb-2002')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
select * from unique where date > '1-Jan-999' 
[continue]
    -> 
Query OK.  2 row(s) modified or retrieved.

 +----------+-------------+
 | key      | date        |
 +----------+-------------+
 | 1        | 01-Jan-1001 |
 | 2        | 02-Feb-2002 |
 +----------+-------------+



mSQL >     ->     -> 
Query buffer
------------

select first_name, last_name from staff order by last_name 
[continue]
    -> 
Query OK.  6 row(s) modified or retrieved.

 +------------+------------+
 | first_name | last_name  |
 +------------+------------+
 | John       | Eales      |
 | Tom        | Hanks      |
 | Mike       | Horn       |
 | David      | Hughes     |
 | Bill       | Jones      |
 | Fred       | Smith      |
 +------------+------------+



mSQL > 
Query buffer
------------
select first_name, last_name from staff order by last_name offset 2
[continue]
    -> 
Query OK.  4 row(s) modified or retrieved.

 +------------+------------+
 | first_name | last_name  |
 +------------+------------+
 | Mike       | Horn       |
 | David      | Hughes     |
 | Bill       | Jones      |
 | Fred       | Smith      |
 +------------+------------+



mSQL > 
Query buffer
------------
select first_name, last_name from staff order by last_name limit 1 offset 3 
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +------------+------------+
 | first_name | last_name  |
 +------------+------------+
 | David      | Hughes     |
 +------------+------------+



mSQL > 
Bye!

